packages/dapp-kit/src/hooks/wallet/useSlushWallet.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { registerSlushWallet } from '@mysten/slush-wallet';
+import { useLayoutEffect } from 'react';
+
+export interface SlushWalletConfig {
+	name: string;
+	network?: 'mainnet' | 'testnet';
+	origin?: string;
+}
+
+export function useSlushWallet(config?: SlushWalletConfig) {
+	useLayoutEffect(() => {
+		if (!config?.name) {
+			return;
+		}
+
+		let cleanup: (() => void) | undefined;
+
+		const setupWallet = async () => {
+			try {
+				const result = await registerSlushWallet(config.name, {
+					origin: config.origin,
+				});
+				cleanup = result?.unregister;
```

Comment:
You need to handle the case where the cleanup hook already has been called but it happened after the register function was called but before you assigned the cleanup function.